### DISCUSSION TITLE
B-tree vs B+tree

### DISCUSSION THREAD

**Discussion by Hanjaemo** [Hanjaemo/DB-Study#30] [Q&A]
DB index로 B+tree를 사용한다고 하는데, B-tree와 B+tree의 차이점은 무엇일까요?

**Comment by gomudayya** [comment]
### B-tree

- 리프노드, 중간노드 모두 실제 데이터(레코드)가 저장됨

- 중간 노드에 실제 데이터와 포인터(인덱스)가 같이 저장됨.

### B+tree

- 리프노드에만 실제 데이터(레코드)가 저장됨. 

- 중간노드 에는 포인터(인덱스)만 저장됨.


범위 검색을 한다고 보면 B+tree가 leaf에 실제 레코드가 전부 있어서 수월한 것 같습니다.

그래서 DB에서도 B+tree를 사용하나 봅니다.
